src/azul/indexer.py

```diff
@@ -19,20 +19,21 @@ class BaseIndexer(ABC):
     """
     def __init__(self, properties: BaseIndexProperties) -> None:
         self.properties = properties
+        self.es_client = ESClientFactory.get()
```

Comment:
I'd prefer to avoid this attribute because `.get()` caches client instances already. To newcomers reading the code I don't want to create the impression that stashing the client as an instance attribute is somehow a efficiency gain. It is but only minuscule one and that's not worth people starting to follow this pattern all over the code base. Local variable is ok, just not an instance attribute.